tools/tplg_parser/tplg_parser.c

```diff
@@ -452,14 +472,21 @@ int tplg_load_pipeline(int comp_id, int pipeline_id, int size,
 		read_size = sizeof(struct snd_soc_tplg_vendor_array);
 		ret = fread(array, read_size, 1, file);
 		if (ret != 1) {
-			free(array);
+			free((void *)array - total_array_size);
```

Comment:
one question, why are we offsetting this array before freeing? Won't any pointer to the malloced region free the whole region?

Reply:
Ok. So, in the function we are incrementing array pointer, there's a chance that the pointer may point to out of bound memory region. So, this is a matter of making sure that it doesn't point to out of memory region.

Reply:
where are we incrementing it? I don't see any increments or any passing by reference of the pointer.

Reply:
At this line: [L480](https://github.com/thesofproject/sof/blob/master/tools/tplg_parser/tplg_parser.c#L480)

Reply:
that is a completely different var though, this memory is only allocated 11 lines above https://github.com/thesofproject/sof/pull/3311/files/1dcaf80b281968d1c6d535a96d897ae03235cabb#diff-cb5e079950432c0d37edc6cbbf0edec1R464

Reply:
nevermind i understand the math now, sorry for the noise